tests/vmi_networking_test.go

```diff
@@ -99,23 +101,31 @@ var _ = Describe("Networking", func() {
 	tests.BeforeAll(func() {
 		tests.BeforeTestCleanup()
 
-		prepareVMIs := func(withPodNetwork bool) (*v1.VirtualMachineInstance, *v1.VirtualMachineInstance) {
+		prepareVMIs := func(withPodNetwork bool) (*v1.VirtualMachineInstance, *v1.VirtualMachineInstance, *v1.VirtualMachineInstance) {
 			// Prepare inbound and outbound VMI definitions
 			inboundVMI := tests.NewRandomVMIWithEphemeralDiskAndUserdata(tests.RegistryDiskFor(tests.RegistryDiskCirros), "#!/bin/bash\necho 'hello'\n")
 			inboundVMI.Labels = map[string]string{"expose": "me"}
 			inboundVMI.Spec.Subdomain = "myvmi"
 			inboundVMI.Spec.Hostname = "my-subdomain"
+
+			inboundVMIWithCustomMacAddress = tests.NewRandomVMIWithEphemeralDiskAndUserdata(tests.RegistryDiskFor(tests.RegistryDiskCirros), "#!/bin/bash\necho 'hello'\n")
+			inboundVMIWithCustomMacAddress.Labels = map[string]string{"expose": "me"}
+			inboundVMIWithCustomMacAddress.Spec.Subdomain = "myvmi"
+			inboundVMIWithCustomMacAddress.Spec.Hostname = "my-subdomain"
+			inboundVMIWithCustomMacAddress.Spec.Domain.Devices.Interfaces[0].MacAddress = "de:ad:00:00:be:af"
```

Comment:
You have to call `v1.SetDefaults_NetworkInterface` before accessing the first interface since `NewRandomVMIWithEphemeralDiskAndUserdata` doesn't add it by default.